### DISCUSSION TITLE
add hosts fail

### DISCUSSION THREAD

**Discussion by ghost** [apache/cloudstack#9155] [Q&A] [answered]
I failed to test adding a host to cloudstack on the local server. I'm not sure what the reason is.

I think my IP address configuration should be correct.

I work on the management platform and host platform under the current server. They are together.

I am using Cloudstack version: 4.19.0.1

My operating system: Debian 12 x64

![image](https://github.com/apache/cloudstack/assets/45416520/b72925d8-b85e-4791-8276-67447fabd98d)

![image](https://github.com/apache/cloudstack/assets/45416520/9da896b6-2285-4fae-af8b-11d21f6e9531)

![image](https://github.com/apache/cloudstack/assets/45416520/dca6307f-b67c-4710-95a2-6892fd50fdc7)

![image](https://github.com/apache/cloudstack/assets/45416520/3fcb10e3-2007-4bcd-b4b1-9c452566a150)

![image](https://github.com/apache/cloudstack/assets/45416520/03f1dbd1-0186-4b3a-b983-c00787376ef0)

![image](https://github.com/apache/cloudstack/assets/45416520/43219eb4-4c0b-4666-b41a-40e9ed2f5eb0)

![image](https://github.com/apache/cloudstack/assets/45416520/10588b05-bce8-4f7e-84b6-a5fe0b84d71e)

![image](https://github.com/apache/cloudstack/assets/45416520/95f38290-fcdb-4250-9673-519e839360ec)

![image](https://github.com/apache/cloudstack/assets/45416520/22338f4e-fa13-488b-9a68-228ae9e9e8c1)

log file
[management-server.log](https://github.com/apache/cloudstack/files/15196380/management-server.log)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by weizhouapache** [comment]
@0xdddddddd 
can you check if file `/dev/kvm` exists ?

**Comment by ghost** [comment]
> @0xdddddddd can you check if file `/dev/kvm` exists ?

Yes, it exists.

![image](https://github.com/apache/cloudstack/assets/45416520/f7cb6d06-8df2-422f-92f7-d50763258910)


![image](https://github.com/apache/cloudstack/assets/45416520/ee010028-099b-4b78-abc1-576dc44b090a)

**Comment by weizhouapache** [comment]
> > @0xdddddddd can you check if file `/dev/kvm` exists ?
> 
> Yes, it exists.
> 
> ![image](https://private-user-images.githubusercontent.com/45416520/327663567-f7cb6d06-8df2-422f-92f7-d50763258910.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ3MTg2NDgsIm5iZiI6MTcxNDcxODM0OCwicGF0aCI6Ii80NTQxNjUyMC8zMjc2NjM1NjctZjdjYjZkMDYtOGRmMi00MjJmLTkyZjctZDUwNzYzMjU4OTEwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTAzVDA2MzkwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTliZDhiNjJhOTk4YjMxZjYzYWI2Yjg4YTAyNzk3NDUxYjZlM2UwM2FhOGExOGVmMDQ4ODhkYWRmNjNlZjFlNTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.QvCt4Q3E2G33QG44kUgVhj6cP7JZ2KvVW0KjtnInEiI)
> 
> ![image](https://private-user-images.githubusercontent.com/45416520/327662015-ee010028-099b-4b78-abc1-576dc44b090a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ3MTg2NDgsIm5iZiI6MTcxNDcxODM0OCwicGF0aCI6Ii80NTQxNjUyMC8zMjc2NjIwMTUtZWUwMTAwMjgtMDk5Yi00Yjc4LWFiYzEtNTc2ZGM0NGIwOTBhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTAzVDA2MzkwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE3MjM3MTYwMGRjYjhjY2RjMTljYzkxZjQ1NDI3Y2M1MGEwMmM2NGI4ZDBmNTAzNmQ0NzAyOGNmY2FkZTkzNzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.bNEXZ4S_jyulfvJWk5jNOJMEwBIMC0QR6e0SO3lSu9s)

that's strange, the logs show
```
Starting to configure your system:
Checking KVM...[Failed]
Please enable KVM on this machine

Try to restore your system:
```

can you re-add the host ?

**Comment by ghost** [comment]
> > > @0xdddddddd can you check if file `/dev/kvm` exists ?
> > 
> > 
> > Yes, it exists.
> > ![image](https://private-user-images.githubusercontent.com/45416520/327663567-f7cb6d06-8df2-422f-92f7-d50763258910.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ3MTg2NDgsIm5iZiI6MTcxNDcxODM0OCwicGF0aCI6Ii80NTQxNjUyMC8zMjc2NjM1NjctZjdjYjZkMDYtOGRmMi00MjJmLTkyZjctZDUwNzYzMjU4OTEwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTAzVDA2MzkwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTliZDhiNjJhOTk4YjMxZjYzYWI2Yjg4YTAyNzk3NDUxYjZlM2UwM2FhOGExOGVmMDQ4ODhkYWRmNjNlZjFlNTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.QvCt4Q3E2G33QG44kUgVhj6cP7JZ2KvVW0KjtnInEiI)
> > ![image](https://private-user-images.githubusercontent.com/45416520/327662015-ee010028-099b-4b78-abc1-576dc44b090a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ3MTg2NDgsIm5iZiI6MTcxNDcxODM0OCwicGF0aCI6Ii80NTQxNjUyMC8zMjc2NjIwMTUtZWUwMTAwMjgtMDk5Yi00Yjc4LWFiYzEtNTc2ZGM0NGIwOTBhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTAzVDA2MzkwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE3MjM3MTYwMGRjYjhjY2RjMTljYzkxZjQ1NDI3Y2M1MGEwMmM2NGI4ZDBmNTAzNmQ0NzAyOGNmY2FkZTkzNzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.bNEXZ4S_jyulfvJWk5jNOJMEwBIMC0QR6e0SO3lSu9s)
> 
> that's strange, the logs show
> 
> ```
> Starting to configure your system:
> Checking KVM...[Failed]
> Please enable KVM on this machine
> 
> Try to restore your system:
> ```
> 
> can you re-add the host ?

Okay, please wait a few minutes. I will clear the logs, add a new host, and provide you with a new log

**Comment by ghost** [comment]
@weizhouapache 

Hello, I have kept you waiting for a long time. I have added a brand new host log, and besides that, I haven't done anything on the web UI.

**Unable to add the host: Cannot find the server resources at http://192.168.64.129**

![image](https://github.com/apache/cloudstack/assets/45416520/e21338fc-8726-483f-8c15-e31e5464630c)


log file:

[agent.log](https://github.com/apache/cloudstack/files/15196913/agent.log)

[management-server.log](https://github.com/apache/cloudstack/files/15196914/management-server.log)

**Comment by ghost** [comment]
@weizhouapache

The libvirtd configuration file, which is automatically generated by Cloudstack, has not been modified by me

[libvirtd.txt](https://github.com/apache/cloudstack/files/15196981/libvirtd.txt)

**Comment by ghost** [comment]
@weizhouapache 

After my continuous investigation. I found that the kvm ok program is missing in the debian12 system. That's because cloudstack setup agent called kvm ok.
Just install: **sudo apt install cpu-checker** to solve the problem.
This is the "/var/log/cloudstack/agent/setup. log" log error.


```
DEBUG:root:Failed to execute:/bin/sh: 1: kvm-ok: not found
DEBUG:root:execute:/usr/bin/lsb_release -i
DEBUG:root:execute:hostname -f
DEBUG:root:execute:kvm-ok
DEBUG:root:Failed to execute:/bin/sh: 1: kvm-ok: not found
DEBUG:root:execute:/usr/bin/lsb_release -i
DEBUG:root:execute:hostname -f
DEBUG:root:execute:kvm-ok
DEBUG:root:Failed to execute:/bin/sh: 1: kvm-ok: not found
DEBUG:root:execute:/usr/bin/lsb_release -i
DEBUG:root:execute:hostname -f
DEBUG:root:execute:kvm-ok
DEBUG:root:Failed to execute:/bin/sh: 1: kvm-ok: not found
```


In addition, I am facing a new problem
Error will be reported when adding storage
"Something wet wrong; Please correct the following: Fail"

**Comment by ghost** [comment]
When I failed to add storage, I clicked on return to fix and then clicked on next again. Add Host reported an error again, stating that it already exists. I think this is a logical bug in the web UI.

![image](https://github.com/apache/cloudstack/assets/45416520/13eb3e66-ebd6-4dd6-b2b9-0e51f5384c77)

**Comment by ghost** [comment]
Cloudstack agent has been successfully added, but it is always in an initialized state.
Because I am testing cloudstack in a virtual machine, which belongs to nested virtualization.
Cloudstack was executing LSCPU related instructions and attempted to obtain CPU related information, but failed. Causing the cloudstack agent to remain in the initialization phase forever.
Ultimately, the storage function could not be added properly.

[agent.log](https://github.com/apache/cloudstack/files/15199379/agent.log)

**Comment by weizhouapache** [comment]
> When I failed to add storage, I clicked on return to fix and then clicked on next again. Add Host reported an error again, stating that it already exists. I think this is a logical bug in the web UI.
> 
> ![image](https://private-user-images.githubusercontent.com/45416520/327740759-13eb3e66-ebd6-4dd6-b2b9-0e51f5384c77.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ3Mzk3NzIsIm5iZiI6MTcxNDczOTQ3MiwicGF0aCI6Ii80NTQxNjUyMC8zMjc3NDA3NTktMTNlYjNlNjYtZWJkNi00ZGQ2LWIyYjktMGU1MWY1Mzg0Yzc3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTAzVDEyMzExMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE1MjAxNGVlZmNkN2YwMDVlNDBiZWJlZTM3ZmFlMGQzYmM4YTYxNDBmYTFkYTk3MjgxMTAzY2FmZjg5NmUxYmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.IPIuoCx75M0IDmuOy52kJqntBjYXpKMWsrAeIg8jvXg)

this is a known UI bug:  https://github.com/apache/cloudstack/issues/8597
please ignore it for now

**Comment by weizhouapache** [comment]
> Cloudstack agent has been successfully added, but it is always in an initialized state. Because I am testing cloudstack in a virtual machine, which belongs to nested virtualization. Cloudstack was executing LSCPU related instructions and attempted to obtain CPU related information, but failed. Causing the cloudstack agent to remain in the initialization phase forever. Ultimately, the storage function could not be added properly.
> 
> [agent.log](https://github.com/apache/cloudstack/files/15199379/agent.log)

agent.log looks ok, although there are some errors.

is the host Up ? check Infrastructure -> Hosts

**Comment by ghost** [comment]
> > Cloudstack agent has been successfully added, but it is always in an initialized state. Because I am testing cloudstack in a virtual machine, which belongs to nested virtualization. Cloudstack was executing LSCPU related instructions and attempted to obtain CPU related information, but failed. Causing the cloudstack agent to remain in the initialization phase forever. Ultimately, the storage function could not be added properly.
> > [agent.log](https://github.com/apache/cloudstack/files/15199379/agent.log)
> 
> agent.log looks ok, although there are some errors.
> 
> is the host Up ? check Infrastructure -> Hosts

No, I don't know where the mistake was, I feel like there's still a long way to go.
The main issue is the lack of some components, but I don't know what they are.

**Comment by weizhouapache** [comment]
> > > Cloudstack agent has been successfully added, but it is always in an initialized state. Because I am testing cloudstack in a virtual machine, which belongs to nested virtualization. Cloudstack was executing LSCPU related instructions and attempted to obtain CPU related information, but failed. Causing the cloudstack agent to remain in the initialization phase forever. Ultimately, the storage function could not be added properly.
> > > [agent.log](https://github.com/apache/cloudstack/files/15199379/agent.log)
> > 
> > agent.log looks ok, although there are some errors.
> > 
> > is the host Up ? check Infrastructure -> Hosts
> 
> No, I don't know where the mistake was, I feel like there's still a long way to go.
> The main issue is the lack of some components, but I don't know what they are.

Add primary storage
Add secondary storage
Enable the zone

**Comment by ghost** [comment]
> > > > Cloudstack agent has been successfully added, but it is always in an initialized state. Because I am testing cloudstack in a virtual machine, which belongs to nested virtualization. Cloudstack was executing LSCPU related instructions and attempted to obtain CPU related information, but failed. Causing the cloudstack agent to remain in the initialization phase forever. Ultimately, the storage function could not be added properly.
> > > > [agent.log](https://github.com/apache/cloudstack/files/15199379/agent.log)
> > > 
> > > 
> > > agent.log looks ok, although there are some errors.
> > > is the host Up ? check Infrastructure -> Hosts
> > 
> > 
> > No, I don't know where the mistake was, I feel like there's still a long way to go.
> > The main issue is the lack of some components, but I don't know what they are.
> 
> Add primary storage Add secondary storage Enable the zone

I can't add it because they will report errors. I've been working on it all day and I'm very tired. There are not enough detailed logs, the failure rate is too high. Poor system compatibility. The official installation package contains too many missing items during installation. Not easy to get started with. I feel very uncomfortable.

**Comment by rajujith** [comment]
@0xdddddddd I tried using Debian12 as a KVM host with CloudStack 4.19.0.1. `Addhost` failed initially, then I installed the `cpu-checker`. Then executed the `cloudstack-setup-agent` , and the host is visible in the cloudstack UI now.

**Comment by ghost** [comment]
> @0xdddddddd I tried using Debian12 as a KVM host with CloudStack 4.19.0.1. `Addhost` failed initially, then I installed the `cpu-checker`. Then executed the `cloudstack-setup-agent` , and the host is visible in the cloudstack UI now.

Although successful, it's just that you can see success, and in fact, the functionality is not available.
The default installation package is missing too many dependent components. Some things are not included in Debian itself, so I think you should check if there are any critical components in the system

**Comment by yadvr** [comment, marked as answer]
@0xdddddddd Yes, it's because the current deb packages are only tested, used and supported for Ubuntu that you're welcome to use to an out-of-box installation experience, support for Debian is largely experimental and it's possible (as you've found out the hard way) Debian needs additional packages and specific configuration.

Most of us may not have any incentive to support Debian (yet), and it's hard to work on all the issues, features and requirements that each and every user wants in our 'free time'. We want to help but as much frustrating it is for you, it's equally or may be more frustrating that we can't help or support you in something that's not supported (yet). As a community-driven opensource project, our community of users help solve such problems with us and we welcome any and all feedback and contributions, and pull requests.
